core/tests/puppeteer-acceptance-tests/user-utilities/curriculum-admin-utils.ts

```diff
@@ -0,0 +1,703 @@
+      await this.clickOn(mobilePublishButton);
+    } else {
+      await this.clickOn(saveChangesButton);
+      await this.clickOn(saveDraftButton);
+      await this.page.waitForSelector(
+        `${publishExplorationButton}:not([disabled])`
+      );
+      await this.clickOn(publishExplorationButton);
+    }
+
+    await this.type(explorationTitleInput, title);
+    await this.type(explorationGoalInput, `${title} Goal`);
+    await this.clickOn(explorationCategoryDropdown);
+    await this.clickOn('Algebra');
+    await this.clickOn(saveExplorationChangesButton);
+    await this.clickOn(explorationConfirmPublishButton);
+    await this.page.waitForSelector(explorationIdElement);
+    const explorationIdUrl = await this.page.$eval(
+      explorationIdElement,
+      element => element.textContent
+    );
+    if (!explorationIdUrl) {
+      throw new Error('Failed to fetch newly created exploration URL.');
+    }
+    return this.getExplorationIdFromUrl(explorationIdUrl);
+  }
+
+  /**
+   * Function for creating a topic in the topics-and-skills dashboard.
+   */
+  async createTopic(name: string, urlFragment: string): Promise<void> {
+    await this.clickOn(addTopicButton);
+    await this.type(topicNameField, name);
+    await this.type(topicUrlFragmentField, urlFragment);
+    await this.type(topicWebFragmentField, name);
+    await this.type(
+      topicDescriptionField,
+      `Topic creation description test for ${name}.`
+    );
+
+    await this.clickOn(photoBoxButton);
+    await this.page.waitForSelector(photoUploadModal, {visible: true});
+    await this.uploadFile(curriculumAdminThumbnailImage);
+    await this.page.waitForSelector(`${uploadPhotoButton}:not([disabled])`);
+    await this.clickOn(uploadPhotoButton);
+    await this.page.waitForSelector(photoUploadModal, {hidden: true});
+    await this.clickOn(createTopicButton);
+
+    // Creating a topic opens a new tab and brings this.page out of focus,
+    // so we close the newly opened tab.
+    const target = await this.browserObject.waitForTarget(
+      target => target.opener() === this.page.target()
+    );
+    const newTab = await target.page();
+    await newTab?.close();
+
+    await this.openTopicEditor();
+    await this.page.waitForSelector(topicMetaTagInput);
+    await this.page.focus(topicMetaTagInput);
+    await this.page.type(topicMetaTagInput, 'meta');
+    await this.page.keyboard.press('Tab');
+    await this.saveTopicDraft();
+  }
+
+  /**
+   * Function that opens the topic editor page for the topic created.
+   */
+  async openTopicEditor(): Promise<void> {
+    await this.navigateToTopicAndSkillsDashboardPage();
+    await this.clickOn(topicsTab);
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(mobileTopicSelector);
+    } else {
+      await this.clickOn(topic);
+    }
+  }
+
+  /**
+   * Function that opens the skill editor page for the topic created.
+   */
+  async openSkillEditor(): Promise<void> {
```

Comment:
Ditto, doesn't this need the skill as an argument? Otherwise if we ever write a test that exercises a scenario with > 1 skills, we cannot disambiguate.